src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,140 @@
 package core.basesyntax;
 
+import edu.emory.mathcs.backport.java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    Object [] elements;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkRange(index);
+        if (elements.length == size) {
+            increaseArraySize();
+        }
+        for (int i = size; i > index; i--) {
+            elements[i] = elements[i - 1];
+        }
```

Comment:
interesting solution, but let's use System.arraycopy() this would be more confident

Reply:
done!